client/my-sites/patterns/components/pattern-library/index.tsx

```diff
@@ -107,6 +99,38 @@ export const PatternLibrary = ( {
 		},
 	} );
 
+	const isLoggedIn = useSelector( isUserLoggedIn );
+	const isDevAccount = useSelector( ( state ) => getUserSetting( state, 'is_dev_account' ) );
+
+	const handleViewChange = useCallback(
+		( view: 'grid' | 'list' ) => {
+			const searchParams = new URLSearchParams( location.search );
+			const currentView = searchParams.get( 'grid' ) === '1' ? 'grid' : 'list';
+
+			// Let's only handle this if the view changes
+			if ( currentView === view ) {
+				return;
+			}
+
+			recordTracksEvent( 'calypso_pattern_library_view_switch', {
+				category,
+				is_logged_in: isLoggedIn,
+				type: patternTypeFilter === PatternTypeFilter.REGULAR ? 'pattern' : 'page-layout',
+				user_is_dev_account: isDevAccount ? '1' : '0',
+				view,
+			} );
+
+			searchParams.delete( 'grid' );
+			if ( view === 'grid' ) {
+				searchParams.set( 'grid', '1' );
+			}
+
+			const paramsString = searchParams.toString().length ? `?${ searchParams.toString() }` : '';
+			page( location.pathname + paramsString );
```

Comment:
```suggestion
			const url = new URL( window.location.href );
			url.searchParams.delete( 'grid' );

			if ( view === 'grid' ) {
				url.searchParams.set( 'grid', '1' );
			}

			page( url.href.replace( url.origin, '' ) );
```
I know this code was here before your PR, but this would still be a good opportunity to tweak it.

I'm about to update `usePatternSearchTerm` in the same way I suggested here. The upside is that the URL hash is also preserved, which is good for the single-pattern permalinks (they are hash-based).

Reply:
Sounds good